src/auth.rs

```diff
@@ -61,6 +62,24 @@ impl IntoResponse for ClaimsError {
 #[derive(Clone, Debug)]
 pub struct Claims(CoreUserInfoClaims);
 
+impl Claims {
+    pub fn has_starred(&self, repo_full_name: &str) -> anyhow::Result<bool> {
+        let mut subject_iter = self.subject().split('|');
+        let user_type = subject_iter
+            .next()
+            .ok_or_else(|| anyhow!("Failed to extract user type from OpenID subject"))?;
+
+        if user_type != "github" {
+            bail!("Cannot get the stars of a non-github user");
+        }
+
+        let user_id = subject_iter
+            .next()
+            .ok_or_else(|| anyhow!("Failed to extract user id from OpenID subject"))?;
+        github::has_starred(user_id, repo_full_name)
+    }
```

Comment:
I don't mind this as temporary solution, but let's file an issue to use the Auth0 API here.
Eventually we will allow different ID providers (e.g. Google), and this won't work for cases when user first logs in with any other identity provider than GitHub and later associates their account with GitHub